### PR TITLE
Update dependencies and clean up TODOs

### PR DESCRIPTION
**Pull request by Josmithr**
Update fluid dependencies, and use `ConnectionState` in locations where we had been using hard-coded numeric values.

See https://github.com/microsoft/FluidFramework/issues/10681

Also fix license in node-demo's package.json to MIT

### CODE REVIEW

File: node-demo/package.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
   },
   "license": "ISC",
```
Comment (tylerbutler) [inline node-demo/package.json:13]: I think this should be MIT. My guess is that the original developer used the default license from `npm init`. Ordinarily I'd suggest splitting it out into a separate change, but I don't think it's worth the extra hassle for this repo.

**Review by tylerbutler — APPROVED** [review]

File: teams-fluid-hello-world/package.json [hunk 2]
```diff
@@ -13,8 +13,8 @@
   },
   "dependencies": {
     "@fluentui/react-northstar": "~0.58.0",
-    "@fluidframework/azure-client": "^0.59.3000",
-    "@fluidframework/test-client-utils": "^0.59.4001",
+    "@fluidframework/azure-client": "^1.0.0",
```
Comment (skylerjokiel) [inline teams-fluid-hello-world/package.json:16]:
```suggestion
    "@fluidframework/azure-client": "^1.0.1",
```

**Review by skylerjokiel — APPROVED** [review]
Looks good overall 😄

File: node-demo/package.json [hunk 3]
```diff
@@ -10,10 +10,10 @@
     "test": "echo \"Error: no test specified\" && exit 1",
     "test:report": "echo No test for this example"
   },
-  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply node-demo/package.json:13]: Can't take credit though - @tylerbutler noticed it, I just made the fix

**Review by skylerjokiel — APPROVED** [review]
🦜

**Pull request merged**